### ISSUE TITLE
[Action Group] Add `label` prop to component for group context

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#6904]
### Summary

A `label` prop should be added to the `action-group` component for context. 

Currently any grouping does not provide context, and thus does not provide a grouping announcement or context to AT users.

### Actual Behavior

No context is provided when using the `action-group` component and grouping child elements, such as `actions`.

### Expected Behavior

`action-group` should allow folks to specify a grouping name using a `label` prop, similar to https://codepen.io/geospatialem/pen/bGmoyPR.

### Reproduction Sample

https://codepen.io/kevindoshier/pen/PoyJvdN

### Reproduction Steps

1. Open the [sample](https://codepen.io/kevindoshier/pen/PoyJvdN)
2. Open a screen reader (VoiceOver, NVDA, or JAWS)
3. Observe there is no context provided to the screen reader on context within the `action-group`/grouping. 
    - When adding a `aria-label` or `label` no change occurs.
4. Open the [proposed sample](https://codepen.io/geospatialem/pen/bGmoyPR)
5. Open/refer to a screen reader (VoiceOver, NVDA, or JAWS)
6. Observe there is context provided to the screen reader on context within the `action-group`/grouping. 

### Reproduction Version

1.3.1

### Relevant Info

Surfaced as part of MapViewer's accessibility audit in early 2022. This did not surface in Calcite's audit in the fall of 2022, perhaps due to testing issues.

### Priority impact

p2 - want for current milestone

### Esri team

ArcGIS Map Viewer

**Comment by geospatialem** [comment]
Workaround: https://codepen.io/geospatialem/pen/bGmoyPR

**Comment by geospatialem** [comment]
Migrated to August for additional considerations with the `action-pad`/`action-bar`'s `bottom-actions` group `label`.

**Comment by geospatialem** [comment]
Verified in `1.7.0-next.9` with JAWS and NVDA by adding `label` to `calcite-action-group`. Can also be achieved for the last group with `actionsEndGroupLabel` in `action-bar` and `action-pad`:

For instance:
```html
<calcite-action-pad actions-end-group-label="More options are available">
  <calcite-action-group label="Editing">
    <calcite-action text="Home" icon="home"></calcite-action>
    <calcite-action text="Styles" icon="add-in-edit"></calcite-action>
  </calcite-action-group>
  <calcite-action text="Tips" icon="lightbulb"></calcite-action>
</calcite-action-pad>
```
